### PR TITLE
Ensure disallowed tags include unicode empty str

### LINKED ISSUE #34: Disallowed tags within anchors causes an AssertionError
Disallowed tags within anchor tags causes an AssertionError in html5lib. This can be reproduced by trying to sanitize 

`<a href="www.nextthought.com"><div>test</div></a>`

### PR DESCRIPTION
**Pull request by jzuech3**
Fixes #34

### CODE REVIEW

File: src/nti/contentfragments/html.py [hunk 1]
```diff
@@ -195,6 +196,7 @@ def __iter__(self):
     def _find_links_in_text(self, token):
         text = token['data']
         text_and_links = self.link_finder.find_links(text) if self.link_finder else ()
+        #if len(text_and_links) == 0 we return None instead of token
```
Comment (jzuech3) [inline src/nti/contentfragments/html.py]: Thoughts on this block? If text_and_links is empty, we enter this block and return `None`. I was not sure if we wanted to return the token by default in this case.

File: CHANGES.rst [hunk 2]
```diff
@@ -5,7 +5,7 @@
 1.6.1 (unreleased)
 ==================
 
-- Nothing changed yet.
+- Ensure disallowed tags nested within anchors do not raise
```
Comment (jamadden) [inline CHANGES.rst]:
```suggestion
- Ensure disallowed tags nested within anchors do not raise.
  See `issue 34 <https://github.com/NextThought/nti.contentfragments/issues/34>`.
```

File: src/nti/contentfragments/html.py (hunk 1 above)
Comment (jamadden) [reply src/nti/contentfragments/html.py]: This is a generator, so I think what happens in that case is that there's simply nothing to iterate over; the generator yields no items.

That would make the comment incorrect.

**Comment by jamadden** [comment]
Interesting how this doesn't change the coverage at all. Did your added test cases raise the AssertionError before the change?

File: src/nti/contentfragments/html.py (hunk 1 above)
Comment (jamadden) [inline src/nti/contentfragments/html.py]:
```suggestion

```

**Comment by jzuech3** [comment]
The coverage probably did not change because of the changes of state and various conditionals. 

The simple disallowed tag case worked before any changes because it went into the `_find_links_in_text` block, which returned an empty generator (thus we did not have a token with an empty str being encoded and erring in html5lib).

The nested-within-anchors case would not enter the `_find_links_in_text` block (line 190) and return our empty-str non-unicode token and err when passed to html5lib.

**Review by jamadden — APPROVED** [review]
LGTM. I too confirmed this fixed the AssertionError on Python 2.7.

**Comment by jzuech3** [comment]
Thanks!

**Pull request merged**